packages/runtime/test/integration/rekey-transaction.ts

```diff
@@ -0,0 +1,348 @@
+        sign: types.SignType.SecretKey,
+        fromAccount: bob.account,
+        fromAccountAddr: alice.address,
+        toAccountAddr: bob.address,
+        amountMicroAlgos: amount,
+        payFlags: { totalFee: fee }
+      };
+
+      runtime.executeTx(txParam);
+
+      // check balance of alice and bob after transfer
+      syncAccounts();
+      const aliceBalanceAfter = alice.balance();
+      const bobBalanceAfter = bob.balance();
+
+      // transaction fee paid by `from account` not `signer`
+      assert.equal(aliceBalanceBefore, aliceBalanceAfter + amount + BigInt(fee));
+      assert.equal(bobBalanceBefore + amount, bobBalanceAfter);
+    });
+
+    it("Should fail because signer account is invalid spend address", function () {
+      txParam = {
+        type: types.TransactionType.TransferAlgo,
+        sign: types.SignType.SecretKey,
+        fromAccount: alice.account,
+        toAccountAddr: bob.address,
+        amountMicroAlgos: amount,
+        payFlags: { totalFee: fee }
+      };
+
+      expectRuntimeError(
+        () => runtime.executeTx(txParam),
+        RUNTIME_ERRORS.GENERAL.INVALID_AUTH_ACCOUNT,
+        rekeyMessageError(alice.getSpendAddr(), alice.address)
+      );
+    });
+  });
+
+  describe("Account to Lsig", function () {
+    this.beforeEach(() => {
+      // create rekey transaction
+      txParam = {
+        type: types.TransactionType.TransferAlgo,
+        sign: types.SignType.SecretKey,
+        fromAccount: alice.account,
+        toAccountAddr: alice.address,
+        amountMicroAlgos: 0,
+        payFlags: { totalFee: fee, rekeyTo: lsigAccount.address }
+      };
+
+      runtime.executeTx(txParam);
+      syncAccounts();
+    });
+
+    it("spend address of alice account should be lsig address", () => {
+      assert.isNotNull(alice.account.spend);
+      assert.equal(alice.getSpendAddr(), lsigAccount.address);
+    });
+
+    it("Transfer ALGO by valid spend account", () => {
+      // balance before rekey
+      const aliceBalanceBefore = alice.balance();
+      const bobBalanceBefore = bob.balance();
+
+      // transfer ALGO use lsig
+      txParam = {
+        type: types.TransactionType.TransferAlgo,
+        sign: types.SignType.LogicSignature,
+        fromAccountAddr: alice.address,
+        toAccountAddr: bob.address,
+        amountMicroAlgos: amount,
+        lsig: lsig,
+        payFlags: { totalFee: fee }
+      };
+
+      runtime.executeTx(txParam);
+
+      // check balance of alice and bob after transfer
+      syncAccounts();
+      const aliceBalanceAfter = alice.balance();
+      const bobBalanceAfter = bob.balance();
+
+      // transaction fee will paid by `from account` not `signer account`
+      assert.equal(aliceBalanceBefore, aliceBalanceAfter + amount + BigInt(fee));
+      assert.equal(bobBalanceBefore + amount, bobBalanceAfter);
+    });
+
+    it("Should failed because cloneLsig is invalid spend address of alice account", () => {
```

Comment:
could you also add a failing test, when trying to send ALGO's from "original" account as signer, when account has been rekeyed to lsig ?